### PR TITLE
Add example for a checkbox in the indeterminate state

### PR DESCRIPTION
**Pull request by jorgemasta**
When the `useCheckbox` tried to get the value for the `checked` variable, it omitted the `mixed` state. The only way to set a checkbox as indeterminate was by using the checked prop:
```jsx
<label>
  <Checkbox state={checkbox} checked={checkbox.value} />
  Fruits
</label>
```

With this PR, I have added an example of a checkbox in the indeterminate (or mixed) state.

### CODE REVIEW

File: packages/ariakit/src/checkbox/__examples__/checkbox-indeterminate/index.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import * as React from "react";
+
+import { Checkbox, useCheckboxState } from "ariakit/checkbox";
```
Comment (diegohaz) [inline packages/ariakit/src/checkbox/__examples__/checkbox-indeterminate/index.tsx]:
```suggestion
import { useEffect } from "react";
import { Checkbox, useCheckboxState } from "ariakit/checkbox";
```

File: packages/ariakit/src/checkbox/__examples__/checkbox-indeterminate/index.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import "./style.css";
+
+function useTreeState({ values }: { values: Array<string | number> }) {
+  const group = useCheckboxState();
+  const items = useCheckboxState<Array<string | number>>({ defaultValue: [] });
+  // updates items when group is toggled
+  React.useEffect(() => {
```
Comment (diegohaz) [inline packages/ariakit/src/checkbox/__examples__/checkbox-indeterminate/index.tsx]:
```suggestion
  const items = useCheckboxState<Array<string | number>>({ defaultValue: [] });

  // updates items when group is toggled
  useEffect(() => {
```

File: packages/ariakit/src/checkbox/__examples__/checkbox-indeterminate/index.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    if (group.value === true) {
+      items.setValue(values);
+    } else if (group.value === false) {
+      items.setValue([]);
+    }
+  }, [group.value]);
+
+  // updates group when items is toggled
+  React.useEffect(() => {
```
Comment (diegohaz) [inline packages/ariakit/src/checkbox/__examples__/checkbox-indeterminate/index.tsx]:
```suggestion
  useEffect(() => {
```

File: packages/ariakit/src/checkbox/__examples__/checkbox-indeterminate/index.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+  return { group, items };
+}
+
+export default function Example() {
+  const values = ["Apple", "Orange", "Mango"];
+
+  const { group, items } = useTreeState({ values });
+
+  return (
```
Comment (diegohaz) [inline packages/ariakit/src/checkbox/__examples__/checkbox-indeterminate/index.tsx]: Nit:
```suggestion
  const values = ["Apple", "Orange", "Mango"];
  const { group, items } = useTreeState({ values });
  return (
```

File: .changeset/checkbox-indeterminate-state.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+"ariakit": minor
+---
+
+Fixed the `getStateChecked` to consider mixed or undetermined states
```
Comment (diegohaz) [inline .changeset/checkbox-indeterminate-state.md]:
```suggestion
Added support for `"mixed"` as a value for the `useCheckboxState`'s `value` prop. ([#2158](https://github.com/ariakit/ariakit/pull/2158))
```

File: packages/ariakit/src/checkbox/checkbox.tsx [hunk 6]
```diff
@@ -66,6 +69,8 @@ export const useCheckbox = createHook<CheckboxOptions>(
       checkedProp ?? getStateChecked(state?.value, value)
     );
 
+    console.log("checked", checked);
+
```
Comment (diegohaz) [inline packages/ariakit/src/checkbox/checkbox.tsx]:
```suggestion

```

**Review by diegohaz — COMMENTED** [review]
Thanks for the contribution @jorgemasta! I made some suggestions.

However, I'm not sure we should support this feature. The `value` prop on `useCheckboxState` currently supports string values, which are checked against the `Checkbox` component's `value` prop. And this special `mixed` value could conflict or, at the very least, be confusing.

If you want a `mixed` value, just set the `checked` prop on `Checkbox` directly.

**Comment by jorgemasta** [comment]
@diegohaz  Thanks for the suggestions and the quick feedback.

You're right. Supporting string values except `mixed` doesn't look like a good idea. But what is the use case of using a string as a value? I understand an array of strings for multiple `Checkbox` elements.

**Comment by diegohaz** [comment]
> But what is the use case of using a string as a value?

I personally don't find it very useful, but there are a lot of checkbox examples that use the `value` attribute as a string out there. This is one of the first examples on the MDN page: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#value

**Comment by jorgemasta** [comment]
Hey @diegohaz, I have removed my changes and kept the example of a checkbox in the indeterminate state.

But I'm still unsure if the `getStateChecked` needs some revisions. For example, if I initialise a checkbox with a sting default value:
```js
const group = useCheckboxState({ defaultValue: "newsletter" });
```
I expected the checkbox to be checked, but it's not the case. The return value from `getStateChecked` is false for any string.

**Pull request closed without merging**